File: 2020/python/day3/day3_1.py
#!/usr/bin/env python3
# Author: Mathias Rahbek-Borre
# Advent of code - day 3 part 1:

infile = open("input.txt", "r")

step_x = 3
step_y = 1
index = 0
tree_count = 0

for inline in infile:
    line = inline.strip()
    if index >= len(line):
        index = index - len(line)
    if line[index] == "#":
        tree_count += 1
    index += step_x
print(tree_count)

File: 2020/cpp/day3/day3_1.cpp
#include <iostream>
#include <fstream>

using namespace std;

int main()
{
    string line;

    ifstream infile("input.txt");

    if (infile.is_open())
    {
        int step_x = 3;

        int trees = 0;
        int index = 0;
        while (getline(infile&, line))
        {
            if (line[index] == '#')
            {
                trees++;
            }
            index += step_x;
            if (index >= line.size())
            {
                index -= line.size();
            }
        }
        infile.close();
        cout << trees << endl;
    }
}